referenceframe/frame.go

```diff
@@ -198,6 +198,22 @@ func (nf *noGeometryFrame) Geometries(input []Input) (*GeometriesInFrame, error)
 	return NewGeometriesInFrame(nf.Name(), nil), nil
 }
 
+// namedFrame is used to change the name of a frame.
+type namedFrame struct {
```

Comment:
> add a Rename function to baseFrame and all other frames will have access to it

This is precisely what I'm trying to prevent with this change :)

I don't want some function somewhere able to change the name of a frame for everything else holding onto it. If a frame changes its name, that should now point to a new thing.

Reply:
A `namedFrame` does not necessarily wrap a `baseFrame`. Whatever the wrapped `Frame` has as a name is essentially gone for all intents and purposes for anything that has a pointer to the named frame, but the wrapped frame is unchanged for anything with a pointer to that.

Having a `ChangeName` function implies we're mutating the frame, which we don't want to do, explicitly or otherwise.

Reply:
Right I see what you are saying but in I think 100% of cases currently we will at the end of the day wrap a baseFrame with a second name, making the first obsolete which I think could be confusing.  Would you feel better if we changed the method to something like `CopyWithNewName`? 

Otherwise I don't think this is a huge deal so I'm good to LGTM if you would rather continue in this direction.

Reply:
This won't make `baseFrame` names obsolete. We don't often rename frames; we have only been doing so historically for models. The use case here is that models store optionally names for their model frames in the kinematics files, which is the original name of the model frame, but this almost-never matches the component name.

The only current change happening here is that that model is now wrapped with the correct frame name, rather than editing the struct.

Reply:
i.e. static frames, etc, will all retain their baseframe name.

IMO `CopyWithNewName` is more misleading since we don't copy, just wrap.